src/org/traccar/reports/Events.java

```diff
@@ -92,7 +92,7 @@ public static void getExcel(OutputStream outputStream,
             deviceEvents.setDeviceName(device.getName());
             sheetNames.add(WorkbookUtil.createSafeSheetName(deviceEvents.getDeviceName()));
             if (device.getGroupId() != 0) {
-                Group group = Context.getDeviceManager().getGroupById(device.getGroupId());
+                Group group = (Group) Context.getGroupsManager().getById(device.getGroupId());
```

Comment:
Can we have a generic function that will automatically return correct object type?